package/gluon-mesh-vpn-wireguard/files/lib/gluon/upgrade/404-wireguard-generate-key

```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+secretfile=/lib/gluon/mesh-vpn/wireguard
+if ! wg pubkey < $secretfile >/dev/null 2>&1
+then
+  echo "generate" > $secretfile
```

Comment:
Why not just make sure to delete the file if it's not a valid secret?

Reply:
this follows the pattern from the fastd implemenation. it allows to get a key generated on startup by the initscript with haveged running instead of in config mode where it is not running.

Reply:
The key should be persisted in UCI and extracted from there when needed as a file.

Reply:
this is done in the latest patchset, running tests and will include the commit here once confirmed working